src/consumers/manager.rs

```diff
@@ -267,12 +267,13 @@ where
     C: Stream<Item = Result<ScopedMessage<W::Message>, async_nats::Error>> + Unpin,
 {
     loop {
+        // Get next value from stream, returning error if the consumer stopped
+        let res = consumer.next().await.ok_or(WorkError::ConsumerStopped)?;
```

Comment:
This unfortunately could also cause issues. Let's say we've reached the limits on permits, now we've pulled work from the queue and could end up waiting until the timeout for the message (or close to it) and then it will expire. So we'll start doing work and then fail when we try to ack (leading to possible duplication). We need to make sure we can a) get a permit and b) that there is work to pull